### DISCUSSION TITLE
Check expire timestamp

### DISCUSSION THREAD

**Discussion by marcus-at-localhost** [nette/caching#63] [Q&A]
I'm trying to cache some data indefinitely and invalidate the cache when data was updated.

I've got the timestamp when the data was updated, but can't compare it to the timestamp when the data was cached. How can I solve this?

I thought of something like this:

```php
$timestamp // from database call

if($timestamp > $cached_data_timestamp) {
  $cache->remove($key);
  $value = $cache->load($key, function () use ($key) {
	$computedValue = ...; // heavy computations
	return $computedValue;
});
}

```

But I just can't know when the cache was written?! (I'm using the sqlite storage, but that shouldn't matter)

One idea was, to use the `$timestamp` as part of the key, but then I would just fill up the cache?

Thanks for any input.

**Comment by dg** [comment]
Nette does not store the time when the cache item was created. You have to add this to the data yourself, ie. `return ['time' => time(), 'value' => $computedValue];`
